### ISSUE TITLE
TypeScript type error cannot assign `unknown` to string...

### ISSUE THREAD

**Issue by ghiscoding** [ghiscoding/slickgrid-universal#1422]
### Describe the bug

I think the upgrade to the new TypeScript version created this new issue but only after I deleted the pnpm lock file and reinstalled everything

https://github.com/ghiscoding/slickgrid-universal/blob/5fab1792cfdf24172bd46556b6fe5513c93d19d1/packages/common/src/interfaces/column.interface.ts#L24-L28

![image](https://github.com/ghiscoding/slickgrid-universal/assets/643976/6b082229-5c75-450c-9820-9db51df7c36b)

### Reproduction

pnpm install and then run a TypeScript build

### Which Framework are you using?

Vanilla / Plain JS

### Environment Info

```shell
n/a
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/slickgrid-universal/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [Wikis](https://github.com/ghiscoding/slickgrid-universal/wiki).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/slickgrid-universal/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/slickgrid-universal/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
@zewa666 can you help with this error since it is code you've put in couple years ago 😉 
I think for now, I'll simply add a `@ts-ignore`

**Comment by ghiscoding** [comment]
alright, I'll let you propose a fix then if you don't mind. Thanks 😉

**Comment by ghiscoding** [comment]
> that sounds really cool. So its a way to quickly review an incoming PR? very cool, i like.

@zewa666 
Yeah I just added Stackblitz app to my GitHub account and a Stackblitz link to my readme page (in this commit eaf3bfd8) and that was it, I was actually surprised about the PR, I just wanted a Stackblitz button (I copied the link from a GitHub project I saw and just tried it for fun but I find it to be very useful). More info on [Stackblitz add PR new button](https://developer.stackblitz.com/codeflow/using-pr-new)

BTW, have you had a chance to try the Dark Mode yet? 😉

**Comment by zewa666** [comment]
no, sadly I'm burried in work with AI stuff currently. But finally begin of April my larger grid project is about to begin so I'm looking forward getting up to speed with all the new features.
